src/components/CopyAdForm/CopyAdForm.tsx

```diff
@@ -0,0 +1,281 @@
+import { RATE_TYPE } from '@/constants';
+import { api, useFloatingRate } from '@/hooks';
+import { AdFormInput } from '@/pages/my-ads/components/AdFormInput';
+import { formatTime, getValidationRules, restrictDecimalPlace } from '@/utils';
+import { Localize, useTranslations } from '@deriv-com/translations';
+import { InlineMessage, Text, useDevice } from '@deriv-com/ui';
+import { FloatingRate } from '../FloatingRate';
+import CopyAdFormDisplayWrapper from './CopyAdFormDisplayWrapper';
+import './CopyAdForm.scss';
+
+type TSavedFormValues = {
+    amount: number;
+    maxOrder: string;
+    minOrder: string;
+    rateValue: string;
+};
+
+type TCopyAdFormProps = NonUndefined<THooks.AdvertiserAdverts.Get>[0] & {
+    formValues: TSavedFormValues;
+} & {
+    isModalOpen: boolean;
+    onClickCancel: (values: TSavedFormValues) => void;
+    onFormSubmit: (values: TSavedFormValues) => void;
+    onRequestClose: () => void;
+};
+
+type TFormValues = {
+    amount: number;
+    'float-rate-offset-limit': string;
+    'max-order': string;
+    'min-order': string;
+    'rate-type-string': string;
+    'rate-value': string;
+};
+
+const CopyAdForm = ({
+    formValues,
+    isModalOpen,
+    onClickCancel,
+    onFormSubmit,
+    onRequestClose,
+    ...rest
+}: TCopyAdFormProps) => {
+    const {
+        account_currency: currency,
+        amount,
+        description,
+        eligible_countries: eligibleCountries = [],
+        local_currency: localCurrency,
+        min_completion_rate: minCompletionRate = 0,
+        min_join_days: minJoinDays = 0,
+        order_expiry_period: orderExpiryPeriod,
+        payment_method_names: paymentMethodNames,
+        rate_display: rateDisplay,
+        type,
+    } = rest;
+    const { floatRateOffsetLimitString, rateType } = useFloatingRate();
+
+    const methods = useForm<TFormValues>({
+        defaultValues: {
+            amount: formValues.amount || (amount ?? 0),
+            'float-rate-offset-limit': floatRateOffsetLimitString,
+            'max-order': formValues.maxOrder,
+            'min-order': formValues.minOrder,
+            'rate-type-string': rateType,
+            'rate-value': formValues.rateValue || (rateType === RATE_TYPE.FLOAT ? '-0.01' : rateDisplay),
+        },
+        mode: 'all',
+    });
+
+    const { data: countryList = {} as TCountryListItem } = api.countryList.useGet();
+    const {
+        control,
+        formState: { isValid },
+        getValues,
+        handleSubmit,
+        trigger,
+    } = methods;
+    const { localize } = useTranslations();
+    const { isMobile } = useDevice();
+
+    const onSubmit = () => {
+        onFormSubmit({
+            amount: getValues('amount'),
+            maxOrder: getValues('max-order'),
+            minOrder: getValues('min-order'),
+            rateValue: getValues('rate-value'),
+        });
+    };
+
+    const labelSize = isMobile ? 'sm' : 'xs';
+    const valueSize = isMobile ? 'md' : 'sm';
```

Comment:
We need to start to consider the tablet view.
```suggestion
    const labelSize = isDesktop ? 'xs' : 'sm';
    const valueSize = isDesktop ? 'sm' : 'md';
```

Reply:
since this uses the same components as the ones in create/edit ad it might break the existing ones.

Reply:
probs we need to create another card to check the responsive sizings for the whole of v2

Reply:
yeah.. also we might need design.. because for the copy ads i tried the mobile design for tablet and it looks really bad..